File: java/src/hashing/HashSet.java
package hashing;

import java.util.LinkedList;

public class HashSet<T> {
    LinkedList<T>[] chains = new LinkedList[1000];

    public boolean find(T obj) {
        LinkedList<T> chain = this.chains[obj.hashCode()];

        if (chain == null)
            return false;

        for (T key : chain)
            if (key.equals(obj))
                return true;

        return false;
    }

    public void add(T obj) {
        if (this.find(obj))
            return;

        LinkedList<T> chain = this.chains[obj.hashCode()];
        if (chain == null) {
            chain = new LinkedList<T>();
            this.chains[obj.hashCode()] = chain;
        }
        chain.add(obj);
    }

    public T remove(T obj) {
        if (!this.find(obj))
            return null;

        LinkedList<T> chain = this.chains[obj.hashCode()];
        chain.remove(obj);

        return obj;
    }
}

File: python/hashing/hashSet.py
class HashSet:
    def __init__(self):
        self.chains = [None] * 1000

    def find(self, obj):
        chain = self.chains[obj.hash()]
        
        if chain is None:
            return False

        for key in chain:
            if key.equals(obj):
                return True

        return False

    def add(self, obj):
        if self.find(obj):
            return
        
        chain = self.chains[obj.hash()]
        if chain is None:
            self.chains[obj.hash()] = chain = []
        chain.append(obj)


    def remove(self, obj):
        if self.find(obj) == False:
            return None

        chain = self.chains[obj.hash()]
        chain.remove(obj)

        return obj
